src/main.rs

```diff
@@ -150,13 +162,27 @@ fn main() -> Result<()> {
                 LAST.with(|last| *last.borrow_mut() = now);
                 let datetime: DateTime<Utc> = now.into();
                 let time = datetime.format("%H%M%S.00");
-                let gpgga = format!(
-                    "$GPGGA,{},{:02}{:010.7},{},{:03}{:010.7},{},4,12,1.3,{:.2},M,0.0,M,1.7,0078",
-                    time, lat_deg, lat_min, lat_dir, lon_deg, lon_min, lon_dir, heightf
-                );
-                let checksum = checksum(gpgga.as_bytes());
-                let gpgga = format!("{}*{:X}\r\n", gpgga, checksum);
-                buf.write_all(gpgga.as_bytes()).unwrap();
+                
+                let message = match &opt.areaid {
+                    Some(areaid) => {
+                        let request_counter = match &opt.request_counter {
+                            Some(rc) => rc.to_string(),
+                            None => String::new()
+                        };
+                        let solution_id = match &opt.sid {
+                            Some(sid) => sid.to_string(),
+                            None => String::new()
+                        };
+                        format!("$PSWTCRA,{},{},,{},", request_counter, areaid, solution_id)
```

Comment:
Please refer to PLEASE NOTE field in the following document:
https://swift-nav.atlassian.net/wiki/spaces/ENG/pages/2412019823/DEVINFRA-987+Bosch+CRA+message+in+ntripping